File: Formula/kdbx.rb
class Kdbx < Formula
  desc "A secure hole for your passwords (Keepass CLI)."
  homepage "https://github.com/daxartio/kdbx"
  version "0.12.0"
  license "MIT"
  if OS.mac?
    if Hardware::CPU.intel?
      url "https://github.com/daxartio/kdbx/releases/download/0.12.0/kdbx-0.12.0-x86_64-apple-darwin.tar.gz"
      sha256 "3c091b151bf14178f0091d1daffe86795b9e4eb5ed257aec3c257c325fedc861"
    elsif Hardware::CPU.arm?
      url "https://github.com/daxartio/kdbx/releases/download/0.12.0/kdbx-0.12.0-aarch64-apple-darwin.tar.gz"
      sha256 "82a7ea48b0d82e4e4835ecc13711e837ad0ef37fb112c9863c563c21e01bfee3"
    end
  elsif OS.linux?
    if Hardware::CPU.intel?
      url "https://github.com/daxartio/kdbx/releases/download/0.12.0/kdbx-0.12.0-x86_64-unknown-linux-musl.tar.gz"
      sha256 "b309df868a207203e62f09a42b024d204c36b113898f36363141032aabe1ebf6"
    elsif Hardware::CPU.arm?
      url "https://github.com/daxartio/kdbx/releases/download/0.12.0/kdbx-0.12.0-aarch64-unknown-linux-musl.tar.gz"
      sha256 "a008ab52ca9c16bd87f312666c6ff9ecbfeb7e5a91c6e043b08ec3d29a229b83"
    end
  end
  def install
    bin.install "kdbx"
  end
  test do
    system "#{bin}/kdbx -V"
  end
end

File: scripts/kdbx.py
import sys

import requests

version = sys.argv[1]
urls = {
    "x86_64-apple-darwin": f"https://github.com/daxartio/kdbx/releases/download/{version}/kdbx-{version}-x86_64-apple-darwin.tar.gz",
    "aarch64-apple-darwin": f"https://github.com/daxartio/kdbx/releases/download/{version}/kdbx-{version}-aarch64-apple-darwin.tar.gz",
    "x86_64-unknown-linux-musl": f"https://github.com/daxartio/kdbx/releases/download/{version}/kdbx-{version}-x86_64-unknown-linux-musl.tar.gz",
    "aarch64-unknown-linux-musl": f"https://github.com/daxartio/kdbx/releases/download/{version}/kdbx-{version}-aarch64-unknown-linux-musl.tar.gz",
}


def get_sha256(url):
    r = requests.get(f'{url}.sha256')
    return r.content.decode('utf-8').split(' ')[0]


checksums = {os: get_sha256(url) for os, url in urls.items()}

with open('Formula/kdbx.rb', 'w') as f:
    f.write(
        f"""class Kdbx < Formula
  desc "A secure hole for your passwords (Keepass CLI)."
  homepage "https://github.com/daxartio/kdbx"
  version "{version}"
  license "MIT"

  if OS.mac?
    if Hardware::CPU.intel?
      url "{urls['x86_64-apple-darwin']}"
      sha256 "{checksums['x86_64-apple-darwin']}"
    elsif Hardware::CPU.arm?
      url "{urls['aarch64-apple-darwin']}"
      sha256 "{checksums['aarch64-apple-darwin']}"
    end
  elsif OS.linux?
    if Hardware::CPU.intel?
      url "{urls['x86_64-unknown-linux-musl']}"
      sha256 "{checksums['x86_64-unknown-linux-musl']}"
    elsif Hardware::CPU.arm?
      url "{urls['aarch64-unknown-linux-musl']}"
      sha256 "{checksums['aarch64-unknown-linux-musl']}"
    end
  end

  def install
    bin.install "kdbx"
  end

  test do
    system "#{{bin}}/kdbx -V"
  end
end
"""
    )
